packages/asset/contracts/libraries/LibOrder.sol

```diff
@@ -0,0 +1,155 @@
+
+        if (rightTakeValue > leftMakeValue) {
+            return _fillLeft(leftMakeValue, leftTakeValue, rightOrder.makeAsset.value, rightOrder.takeAsset.value);
+        }
+        return _fillRight(leftOrder.makeAsset.value, leftOrder.takeAsset.value, rightMakeValue, rightTakeValue);
+    }
+
+    /// @notice Computes the remaining fillable amount of an order.
+    /// @param order The order to compute from.
+    /// @param fill The amount of the order already filled.
+    /// @return makeValue The remaining fillable amount from the maker's side.
+    /// @return takeValue The remaining fillable amount from the taker's side.
+    function calculateRemaining(LibOrder.Order calldata order, uint256 fill)
+        internal
+        pure
+        returns (uint256 makeValue, uint256 takeValue)
+    {
+        require(order.takeAsset.value >= fill, "filling more than order permits");
+        takeValue = order.takeAsset.value - fill;
+        makeValue = LibMath.safeGetPartialAmountFloor(order.makeAsset.value, order.takeAsset.value, takeValue);
+    }
+
+    /// @notice Computes the fill values for a situation where the right order is expected to fill the left order.
+    /// @param leftMakeValue The amount the left order maker wants to trade.
+    /// @param leftTakeValue The amount the left order taker wants in return.
+    /// @param rightMakeValue The amount the right order maker wants to trade.
+    /// @param rightTakeValue The amount the right order taker wants in return.
+    /// @return The filled amounts for both the left and right orders.
+    function _fillRight(
+        uint256 leftMakeValue,
+        uint256 leftTakeValue,
+        uint256 rightMakeValue,
+        uint256 rightTakeValue
+    ) internal pure returns (FillResult memory) {
+        uint256 makerValue = LibMath.safeGetPartialAmountFloor(rightTakeValue, leftMakeValue, leftTakeValue);
+        require(makerValue <= rightMakeValue, "fillRight: unable to fill");
+        return FillResult(rightTakeValue, makerValue);
+    }
+
+    /// @notice Computes the fill values for a situation where the left order is expected to fill the right order.
+    /// @param leftMakeValue The amount the left order maker wants to trade.
+    /// @param leftTakeValue The amount the left order taker wants in return.
+    /// @param rightMakeValue The amount the right order maker wants to trade.
+    /// @param rightTakeValue The amount the right order taker wants in return.
+    /// @return The filled amounts for both the left and right orders.
+    function _fillLeft(
+        uint256 leftMakeValue,
+        uint256 leftTakeValue,
+        uint256 rightMakeValue,
+        uint256 rightTakeValue
+    ) internal pure returns (FillResult memory) {
+        uint256 rightTake = LibMath.safeGetPartialAmountFloor(leftTakeValue, rightMakeValue, rightTakeValue);
+        require(rightTake <= leftMakeValue, "fillLeft: unable to fill");
+        return FillResult(leftMakeValue, leftTakeValue);
+    }
+}
```

Comment:
At some point, we will have to take all those interfaces from the marketplace package